src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -181,6 +183,124 @@ public Generator(Logger logger, Properties properties, String encoding) {
+	out.println("");
+	out.println("typedef struct constructor {");
+	out.println("        jclass clazz;");
+	out.println("        jmethodID ctor_mid;");
+	out.println("} constructor;");
+	out.println("");
+	out.println("typedef std::unordered_map<std::type_index, constructor> JavaCPP_JavaCPP_downcast_cache_t;");
+	out.println("");
+	out.println("JavaCPP_JavaCPP_downcast_cache_t JavaCPP_downcast_cache;");
+	out.println("");
+	out.println("extern \"C++\" template<class BASE> jobject JavaCPP_downcast(JNIEnv* env, BASE* obj) {");
+	out.println("    std::type_index id = typeid(*obj);");
+	out.println("    JavaCPP_JavaCPP_downcast_cache_t::iterator ctor = JavaCPP_downcast_cache.find(id);");
+	out.println("    if (ctor != JavaCPP_downcast_cache.end()) {");
+	out.println("        jobject ret = (jobject)env->NewObject(ctor->second.clazz, ctor->second.ctor_mid, NULL);");
+	out.println("        if (ret == NULL || env->ExceptionCheck()) {");
+	out.println("            std::cerr <<  \"Error calling Pointer constructor \" << std::endl;");
+	out.println("            return NULL;");
+	out.println("            }");
+	out.println("        if (JAVACPP_DEBUG_DCAST) std::cerr << \"Success! \" << std::endl;");
+	out.println("        return ret;");
+	out.println("        }");
+	out.println("    else");
+	out.println("        return NULL;");
+	out.println("}");
+	out.println("");
+	out.println("void JavaCPP_dcast_preload(JNIEnv* env, std::string class_name, std::type_index type_id) {");
+	out.println("    if (JAVACPP_DEBUG_DCAST) std::cerr << \"Try to find class \" <<  class_name << std::endl;");
+	out.println("");
+	out.println("    jclass lclazz = env->FindClass(class_name.c_str());");
+	out.println("    if (lclazz == NULL || env->ExceptionCheck()) {");
+	out.println("        std::cerr << \"Error loading class \" <<  class_name.c_str() << std::endl;");
+	out.println("        return;");
+	out.println("        }");
+	out.println("");
+	out.println("    jclass gclazz = (jclass)env->NewGlobalRef(lclazz);");
+	out.println("    if (gclazz == NULL || env->ExceptionCheck()) {");
+	out.println("        std::cerr << \"Error creating class \" <<  class_name  << std::endl;");
+	out.println("        return;");
+	out.println("        }");
+	out.println(" ");
+	out.println("    if (JAVACPP_DEBUG_DCAST) std::cerr << \"Try to call constructor for class \" <<  class_name << std::endl;");
+	out.println("    jmethodID ctor_mid = env->GetMethodID(gclazz, \"<init>\", \"(Lorg/bytedeco/javacpp/Pointer;)V\");");
+	out.println("    if (ctor_mid == NULL || env->ExceptionCheck()) {");
+	out.println("        std::cerr <<  \"Error getting Pointer constructor \" <<  class_name  << std::endl;");
+	out.println("        return;");
+	out.println("        }");
+	out.println("    JavaCPP_downcast_cache[type_id] = constructor{gclazz,ctor_mid};");
+	out.println("    }");
+}
+
+void outputDowncastCache(LinkedHashSet<Class> allClasses) {
+	var baseClasses = new LinkedHashSet<String>();
```

Comment:
Please limit yourself to features available in Java SE 7. Syntactic sugar like this isn't worth losing backward compatibility.